src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -201,7 +270,8 @@ private Dataset retrieveWorker(DatasetSummary summary) {
     public DatasetSummary retrieveSummaryById(UUID id) {
         try {
             String sql = "SELECT " +
-                "id, name, description, default_profile_id, additional_profile_ids, created_date, flightid " +
+                "id, name, description, default_profile_id, additional_profile_ids, created_date, " +
+                "flightid, sharedlock " +
```

Comment:
Why does the shared lock need to be in the DatasetSummary?

It looks like the getter is only used in tests. I wonder if there is a more narrow way of supporting the tests without carting the shared lock around in the runtime system.

Reply:
I could imagine this being useful in certain situations where you might want to take some action based on values in the shared lock but it would be great to get clarification.

Reply:
It looks like both the exclusive and shared lock getter methods were only getting called from the tests. I removed them from the DatasetSummary and made protected getter methods instead. Those protected methods can be accessed from the tests in the same package.

Other than not needing them anywhere now, I don't see a problem with adding the locks into the dataset summary though, so if we do need them in the code down the line (e.g. logging, ...?), then I think this is the right place for them.

Reply:
I don't think we need to store lock state in any of the in-memory data structures. We need it inside the DAO, but we never use it outside the DAO save for testing.

My understanding is that the primary purpose of DatasetSummary is to provide the ingredients to be returned on REST API calls. We are not exposing lock state there, so don't need it there.

It is not a big deal either way, though. Do what you think best!